pyretis feedstock?

EnricoRiccardi opened the pull request:
this is what I am missing to have our code (pyretis) to conda?

thanks

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/pyretis```) and found some lint.

Here's what I've got...


For **recipes/pyretis**:

 * Failed to even lint the recipe (might be a conda-smithy bug) :cry:

EnricoRiccardi commented:
retry

jjhelmus commented:
@EnricoRiccardi This pull request has 695 files added to this repository which I believe is causing some issues. Is there a reason why you need to include all these files?  Typically adding a single `meta.yaml` files to a subdirectory in the recipes folder is all that is needed. All of these files also makes it very difficult to review this PR as the `meta.yaml` cannot be viewed though the GitHub website (at least I cannot).

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```pyretis```) and found some lint.

Here's what I've got...


For **pyretis**:

 * Failed to even lint the recipe (might be a conda-smithy bug) :cry:

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/pyretis```) and found some lint.

Here's what I've got...


For **recipes/pyretis**:

 * Failed to even lint the recipe (might be a conda-smithy bug) :cry:

EnricoRiccardi commented:
@jjhelmus thanks a lot first for you answer and trying to help me. I took away all the non necessary files. My impression is that there must be something extremely obvious that I don't get and that block everything.

I tried many combos to try to make it work, moved the code from gitlab got github, changed metafile, changed the position of the metafile... dunno what to try... always getting the same error..

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```pyretis```) and found some lint.

Here's what I've got...


For **pyretis**:

 * Failed to even lint the recipe (might be a conda-smithy bug) :cry:

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+#  name: {{ name|lower }}
+#  version: {{ version }}
+
+source:
```
Is it possible to use the source code from pypi.io (0.9.2.dev2)? See:

https://conda-forge.org/docs/meta.html#build-from-tarballs-not-repos

EnricoRiccardi replied:
I tried. Still no linting

EnricoRiccardi replied:
ok, seems that I finally managed to make it work. Thank you tons. 

I would like to add mdtraj in the list of packages, but it fails if I do so. Do you have any tips for that?

Once more, really thanks!

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+#  preserve_egg_dir: True
+#  number: 0
+#  skip: True  # [ not py2k or win #]
+  script: python setup.py install 
```
It looks like the `setup.py` uses `setuptools` so try out:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L26

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+  script: python setup.py install 
+
+requirements:
+  build:
```
If you use the `--single-version-externally-managed` flag in the `script` entry mentioned above, you should only need `python` and `setuptools` as build requirements.

EnricoRiccardi replied:
if I do this, wouldn't I loose all the conda checks for package compatibility? Shall I take out setuptool from setup.py?

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+    - tqdm
+    - colorama
+
+  run:
```
If any of the version restrictions are important, please add them:

https://gitlab.com/pyretis/pyretis/blob/master/setup.py#L74-80

EnricoRiccardi replied:
not yet, I still need to be able to upload the first version

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+
+about:
+  home: http://www.pyretis.org
+  license: LGPL 2.1 or later
```
Please include a `license_file`. Also take a look at the example for other fields to populate:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L44-L65

EnricoRiccardi replied:
I am afraid to add at this moment other error sources. I would like to first understand what is going wrong.  I am starting to consider to not use conda any longer, I am loosing too much time and other people, like you, too... :(

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+about:
+  home: http://www.pyretis.org
+  license: LGPL 2.1 or later
+summary: ''
```
You definitely need to add yourself as a maintainer. See:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L67-L72

EnricoRiccardi replied:
done

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```pyretis```) and found some lint.

Here's what I've got...


For **pyretis**:

 * The summary item is expected in the about section.
 * The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```pyretis```) and found some lint.

Here's what I've got...


For **pyretis**:

 * The summary item is expected in the about section.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```pyretis```) and found some lint.

Here's what I've got...


For **pyretis**:

 * The summary item is expected in the about section.
 * The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```pyretis```) and found some lint.

Here's what I've got...


For **pyretis**:

 * Failed to even lint the recipe (might be a conda-smithy bug) :cry:

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```pyretis```) and found some lint.

Here's what I've got...


For **pyretis**:

 * The summary item is expected in the about section.
 * The recipe `license` should not include the word "License".

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```pyretis```) and found it was in an excellent condition.

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+  ## preserve_egg_dir: True
+
+requirements:
+  build:
```
It looks like only `python` and `setuptools` are required to build the package (the `--single-version-externally-managed` flag, means that `install_requires` is ignored).

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+about:
+  home: http://www.pyretis.org
+  license: LGPL-2.1 or later 
+  summary: 'A rare event code based for Transition Interface Sampling'
```
Could you add info for `license_file`, and `dev_url` to the `about` section?

synapticarbors commented on pyretis/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+
+test:
+  imports:
+  - pyretis
```
If any of the command line scripts have some sort of `-h`/`--help` flag that you could use to test to make sure those are in place, please add them as in:

https://conda.io/docs/building/meta-yaml.html#test-commands

jjhelmus commented:
Before merging this PR these commits should be squashed.  If they are not this will add a lot of bulk to the repository which everyone who forks will need to download.

EnricoRiccardi commented:
Erased the commit history
